pinot-common/src/main/java/org/apache/pinot/common/function/TransformFunctionType.java

```diff
@@ -271,6 +271,7 @@ public enum TransformFunctionType {
       OperandTypes.family(ImmutableList.of(SqlTypeFamily.ARRAY)), "vector_norm"),
 
   ARRAY_VALUE_CONSTRUCTOR("arrayValueConstructor", "array_value_constructor"),
+  ARRAY_GENERATE("arrayGenerate", "array_generate"),
```

Comment:
what's the goal we want to achieve here? ARRAY_GENERATE unlike ARRAY_VALUE_CONSTRUCTOR is not a SqlStdOperatorTable entry so it will not work with multi-stage simply by adding this alone

Reply:
I've removed this.

Reply:
removing this entry will make your transform function implementation unable to register (thus unable to run / dead code). 

i would suggest wait until #12302 to be checked in b/c the type system implication is complex for ARRAY_GEN
also another note to reviewer --> histogram does support a weird array gen. let's see if we can merge the 2

Reply:
Okay